### PR TITLE
Promote pending cm deprecations to full deprecations

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is the next stage of #20853.

## PR Checklist

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [n/a] New plotting related features are documented with examples.

**Release Notes**
- [n/a] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [n/a] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [x] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/cm.py [hunk 1]
```diff
@@ -222,7 +222,6 @@ def get_cmap(self, cmap):
 
 @_api.deprecated(
     '3.6',
```
Comment (dstansby) [inline lib/matplotlib/cm.py:224]: Should the version be bumped to 3.7 when the state has changed from pending to non-pending?

**Review by dstansby — APPROVED** [review]
Just one small non-blocking question

File: lib/matplotlib/cm.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/cm.py:224]: Oops, yes, that is true.

File: lib/matplotlib/cm.py [hunk 2]
```diff
@@ -301,19 +297,15 @@ def _get_cmap(name=None, lut=None):
 get_cmap = _api.deprecated(
     '3.6',
```
Comment (ksunden) [inline lib/matplotlib/cm.py:298]:
```suggestion
    '3.7',
```

This one seems to have escaped the bumping to 3.7 that the other ones got

**Pull request merged**